core/src/main/scala/com.snowplowanalytics.stream/loader/Config.scala

```diff
@@ -38,74 +38,61 @@ import pureconfig.module.enumeratum._
 
 object Config {
 
-  implicit def hint[T]: ProductHint[T]              = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
-  implicit val queueHint: FieldCoproductHint[Queue] = new FieldCoproductHint[Queue]("enabled")
-
-  sealed trait StreamType extends EnumEntry with EnumEntry.Hyphencase
-  object StreamType extends Enum[StreamType] {
-    case object Good      extends StreamType
-    case object Bad       extends StreamType
-    case object PlainJson extends StreamType
-
-    val values = findValues
-  }
-
-  sealed trait BadSink extends EnumEntry with EnumEntry.Hyphencase
-  object BadSink extends Enum[BadSink] {
-    case object Stderr  extends BadSink
-    case object Nsq     extends BadSink
-    case object None    extends BadSink
-    case object Kinesis extends BadSink
-
-    val values = findValues
-  }
-
-  sealed trait GoodSink extends EnumEntry with EnumEntry.Hyphencase
-  object GoodSink extends Enum[GoodSink] {
-    case object Elasticsearch extends GoodSink
-    case object Stdout        extends GoodSink
-
-    val values = findValues
-
-    implicit val goodSinkReader: ConfigReader[GoodSink] = deriveEnumerationReader[GoodSink]
-  }
-
-  case class SinkConfig(good: GoodSink, bad: BadSink)
-
-  object SinkConfig {
-    implicit val sinkConfigReader: ConfigReader[SinkConfig] = deriveReader[SinkConfig]
+  implicit def hint[T]: ProductHint[T]               = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+  implicit val queueHint: FieldCoproductHint[Source] = new FieldCoproductHint[Source]("type")
+
+  case class Monitoring(snowplow: Monitoring.SnowplowMonitoring)
+  object Monitoring {
+    implicit val monitoringConfigReader: ConfigReader[Monitoring] =
+      deriveReader[Monitoring]
+
+    case class SnowplowMonitoring(
+      collector: String,
+      appId: String
+    )
+
+    object SnowplowMonitoring {
+      implicit val snowplowMonitoringConfig: ConfigReader[SnowplowMonitoring] =
+        deriveReader[SnowplowMonitoring]
+    }
   }
 
-  case class AWSConfig(accessKey: String, secretKey: String)
-
-  object AWSConfig {
-    implicit val awsConfigReader: ConfigReader[AWSConfig] = deriveReader[AWSConfig]
-  }
+  sealed trait Source extends Product with Serializable
+  object Source {
+    case class Stdin() extends Source
```

Comment:
I guess there is no easy way to make this a case _object_ and still to make the ConfigReader work nicely?  That's fine if there's no nice way to do it.

Reply:
I looked a bit but couldn't find a way to do that. I can try a bit more to find a way.